src/libraries/System.Security.Cryptography.X509Certificates/tests/CollectionTests.cs

```diff
@@ -1640,6 +1690,71 @@ private static void TestExportStore(X509ContentType ct)
             }
         }
 
+        private static void AssertPemExport(X509Certificate2Collection collection, string expectedContents)
+        {
+            Span<char> buffer = new char[expectedContents.Length];
+            int written;
+
+            // If we expect something to get written, try writing to a buffer that is too small by one
+            // and make sure it fails.
+            if (!buffer.IsEmpty)
+            {
+                Assert.False(collection.TryExportCertificatePems(buffer.Slice(1), out written), nameof(collection.TryExportCertificatePems));
+                Assert.Equal(0, written);
+            }
+
+            Assert.True(collection.TryExportCertificatePems(buffer, out written), nameof(collection.TryExportCertificatePems));
+            Assert.Equal(written, buffer.Length);
+            Assert.Equal(expectedContents, new string(buffer));
+
+            string exported = collection.ExportCertificatePems();
+            Assert.Equal(expectedContents, exported);
+        }
+
+        private static void AssertPkcs7PemExport(X509Certificate2Collection collection)
+        {
+            if (PlatformDetection.UsesMobileAppleCrypto)
+            {
+                return;
+            }
+
+            static void AssertPem(X509Certificate2Collection expected, ReadOnlySpan<char> pemActual)
+            {
+                PemFields fields = PemEncoding.Find(pemActual);
+                ReadOnlySpan<char> actualBase64 = pemActual[fields.Base64Data];
+                Assert.Equal("PKCS7", new string(pemActual[fields.Label]));
+                byte[] data = Convert.FromBase64String(new string(actualBase64));
+
+                using ImportedCollection imported = Cert.Import(data);
+                Assert.Equal(expected.ToArray(), imported.Collection.ToArray(), new X509Certificate2EqualityComparer());
+            }
```

Comment:
Can/should we add a test that checks that the PEM preEB started at index 0 (which might require testing that the label was at index 6), and that the postEB ended at `output[^1]` (which, again, might require offset math)?

Basically "did we export one PEM, no trailing newline, and nothing else? (no gunk in the header, no gunk in the footer)"

Reply:
`PemFields` has `Location` which includes the encapsulation boundaries, so this is relatively easy to test.